bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/usage/ClosedWorkload.java

```diff
@@ -70,7 +70,7 @@ private void stopUsers(final int count) {
 
     private void startUsers(final int count) {
         for (int i = 0; i < count; i++) {
-            final ClosedWorkloadUser user = (ClosedWorkloadUser) userFactory.createUser();
+            final ClosedWorkloadUser user = (ClosedWorkloadUser) userFactory.createUser(null);

```

Comment:
For newly introduced parameters and interfaces I would vote to NOT use null as a valid value. Instead, if the processMonitor is optional, use the appropriate Optinal class. You can have the method without parameter delegate by providing Empty. I'd even recommend doing this as a default method in the interface.

Reply:
Indeed there should only exist a single instance of class UserProcessCounterMonitor per simulation. Currently it is created for each single workload. suggested fix: create single instance in simulizar and pass as constructor argument to the respective classes instead of creating in OpenWorkload class (i.e. once per workload)